tests/testthat/test-formatters_var_labels.R

```diff
@@ -20,8 +21,72 @@ testthat::test_that("col_labels returns a vector of column names when fill = TRU
   )
 })
 
+
+# col_labels ----
+testthat::test_that("col_labels<- value accepts character vector", {
+  iris_df <- utils::head(iris, 2)
+  testthat::expect_error(
+    col_labels(iris_df) <- 1:5,
+    "Assertion on 'value' failed: Must be of type 'character'"
+  )
+  testthat::expect_no_error(col_labels(iris_df) <- as.character(1:5))
+})
+
+testthat::test_that("col_labels<- value must be same length as x", {
+  iris_df <- utils::head(iris, 2)
+  testthat::expect_error(
+    col_labels(iris_df) <- as.character(1:4),
+    "Assertion on 'Length of value is equal to the number of columns' failed"
+  )
+})
+
+testthat::test_that("col_labels<- value accepts named vector", {
+  iris_df <- utils::head(iris, 2)
+  testthat::expect_no_error(col_labels(iris_df) <- stats::setNames(as.character(1:5), names(iris)))
+})
+
+testthat::test_that("col_labels<- value names must be same as variable names", {
+  iris_df <- utils::head(iris, 2)
+  testthat::expect_error(
+    col_labels(iris_df) <- stats::setNames(as.character(1:5), toupper(names(iris_df))),
+    "undefined columns selected"
```

Comment:
please make an assertion in the col_labels/col_relabel to avoid this unclear message